pkg/generate/app/sourcelookup.go

```diff
@@ -378,7 +381,7 @@ func StrategyAndSourceForRepository(repo *SourceRepository, image *ImageRef) (*B
 		Binary: repo.binary,
 	}
 
-	if repo.Info() != nil && repo.Info().Dockerfile != nil {
+	if (repo.ignoreRepository || repo.forceAddDockerfile) && repo.Info() != nil && repo.Info().Dockerfile != nil {
```

Comment:
>>    Names are names - they get used a small fraction of the time. Worrying
>>    about names too much is just as bad as worrying not enough about names.

> So let's not bikeshed on this. I'm willing to clean this up in a follow up.

that's a hilarious quote given how many times he has forced name changes on fields.  i'll have to remember that one.

also the whole reason this regression occurred is because of the poorly named "ignoreRepository" field.  If it were named better, I bet this issue would not have occurred when you were making your changes.

So I'd rather not underestimate the importance of good names.

but i'm fine with forceAddDockerfile at this point.